File: templates/node-lambda/test/handler.test.js
const { describe, it } = require("node:test");
const assert = require("node:assert");
const { handler } = require("../src/handler");

describe("handler", () => {
  it("should greet with provided name", async () => {
    const result = await handler({ name: "world" });
    assert.deepStrictEqual(result, { message: "Hello, world!" });
  });

  it("should default to world", async () => {
    const result = await handler({});
    assert.deepStrictEqual(result, { message: "Hello, world!" });
  });
});

File: templates/java-lambda/src/test/java/com/example/HandlerTest.java
package com.example;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.Map;
import org.junit.jupiter.api.Test;

class HandlerTest {
  @Test
  void handlerReturnsGreeting() {
    Handler handler = new Handler();
    Map<String, Object> event = Map.of("name", "world");
    Map<String, Object> result = handler.handleRequest(event, null);
    assertEquals("Hello, world!", result.get("message"));
  }
}
